src/lro/requestUtils.ts

```diff
@@ -0,0 +1,109 @@
+import { HttpOperationResponse } from "@azure/core-http";
+import { terminalStates } from "./constants";
+import { LROResponseInfo } from "./models";
+
+/**
+ * We need to selectively deserialize our responses, only deserializing if we
+ * are in a final LRO response, not deserializing any polling non-terminal responses
+ */
+export function shouldDeserializeLRO(finalStateVia?: string) {
+  let initialOperationInfo: LROResponseInfo | undefined;
+  let isInitialRequest = true;
+
+  return (response: HttpOperationResponse) => {
+    if (response.status < 200 || response.status >= 300) {
+      return true;
+    }
+
+    if (!initialOperationInfo) {
+      initialOperationInfo = getLROData(response);
+    } else {
+      isInitialRequest = false;
+    }
+
+    if (
+      initialOperationInfo.azureAsyncOperation ||
+      initialOperationInfo.operationLocation
+    ) {
+      return (
+        !isInitialRequest &&
+        isAsyncOperationFinalResponse(
+          response,
+          initialOperationInfo,
+          finalStateVia
+        )
+      );
+    }
+
+    if (initialOperationInfo.location) {
+      return isLocationFinalResponse(response);
+    }
+
+    if (initialOperationInfo.requestMethod === "PUT") {
+      return isBodyPollingFinalResponse(response);
+    }
+
+    return true;
+  };
+}
+
+function isAsyncOperationFinalResponse(
+  response: HttpOperationResponse,
+  initialOperationInfo: LROResponseInfo,
+  finalStateVia?: string
+): boolean {
+  const status: string = response.parsedBody?.status || "Succeeded";
+  if (!terminalStates.includes(status.toLowerCase())) {
+    return false;
+  }
+
+  if (initialOperationInfo.requestMethod === "DELETE") {
+    return true;
+  }
+
+  if (
+    initialOperationInfo.requestMethod === "PUT" &&
+    finalStateVia &&
+    finalStateVia.toLowerCase() === "azure-asyncOperation"
```

Comment:
```suggestion
    finalStateVia.toLowerCase() === "azure-asyncoperation"
```
I think this change is needed since `finalStateVia` will always be all lowercase.

Reply:
oh good catch!!!